src/escrows/DolaFraxPyUSDEscrow.sol

```diff
@@ -0,0 +1,239 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+import {YearnVaultV2Helper, IYearnVaultV2} from "src/util/YearnVaultV2Helper.sol";
+import {IRewardPool} from "src/interfaces/IRewardPool.sol";
+import {IConvexBooster} from "src/interfaces/IConvexBooster.sol";
+import {IVirtualBalanceRewardPool} from "src/interfaces/IVirtualBalanceRewardPool.sol";
+
+// StakingWrapper interface for pools with pid 151+
+interface IStakingWrapper {
+    function token() external returns (address);
+}
+
+contract DolaFraxPyUSDEscrow {
+    using SafeERC20 for IERC20;
+
+    error AlreadyInitialized();
+    error OnlyMarket();
+    error OnlyBeneficiary();
+    error OnlyBeneficiaryOrAllowlist();
+
+    uint256 public constant pid = 317;
+
+    IRewardPool public constant rewardPool =
+        IRewardPool(0xE8cBdBFD4A1D776AB1146B63ABD1718b2F92a823);
+    IConvexBooster public constant booster =
+        IConvexBooster(0xF403C135812408BFbE8713b5A23a04b3D48AAE31);
+    // Created while testing, not existing on mainnet yet
+    IYearnVaultV2 public constant yearn =
+        IYearnVaultV2(0x5b737CC835c29c493845353e0399B408Fbd7105D);
+    IERC20 public constant cvx =
+        IERC20(0x4e3FBD56CD56c3e72c1403e103b45Db9da5B9D2B);
+    IERC20 public constant crv =
+        IERC20(0xD533a949740bb3306d119CC777fa900bA034cd52);
+
+    address public market;
+    IERC20 public token;
+    uint256 public stakedBalance;
```

Comment:
It may be worth dropping the `stakedBalance` counter if it's possible to deposit to convex on behalf of the contract. Otherwise adding a function call that allows reading the actual balance of the staking contract and updating `stakedBalance` accordingly may be a good addition.